### ISSUE TITLE
YOLOX: Exceeding YOLO Series in 2021

### ISSUE THREAD

**Issue by Arata2887** [Arata2887/issue_copy#7]
1.どんな論文? /What does this work do?

- https://arxiv.org/abs/2107.08430
- Present some experienced improvements to YOLO series, forming a new high-performance detector—YOLOX.
- Simplify the OTA(an advanced label assignment, which was Presented by the same authors team) to dynamic top-k strategy, named SimOTA, to get an approximate solution.

2.先行研究と比べてどこがすごい? /What makes this work greater than existing works?

![Figure 1](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/3fd6fc91-b306-412f-8fee-30237fbe66f4)

- YOLOX-L achieves 50.0% AP on COCO with 640x640 resolution, outperforming the counterpart YOLOv5-L by 1.8% AP. On models of small size, YOLOX-Tiny and YOLOX-Nano outperform the corresponding counterparts YOLOv4-Tiny and NanoDet3 by 10% AP and 1.8% AP, respectively. 

3.技術や手法のキモはどこ? /What is the heart of this technology or method?

Compared with YOLOv5, YOLOX is technically different in three points:**Decoupled head**; **Anchor-free**; **SimOTA**.

(1) Decoupled head

- YOLOX replace the YOLO detect head with a lite decoupled head as in Fig. 2. Concretely, it contains a 1x1 conv layer to reduce the channel dimension, followed by two parallel branches with two 3x3 conv layers respectively.
![Figure 2](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/c760dc3e-dbee-4638-86b7-7a2052096e72)

- By replacing YOLO’s head with a decoupled one, the converging speed is greatly improved.
![Figure 3](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/45dec83f-7e18-46c8-aa2e-b1ac3c5b590f)

(2) Anchor-free

- Anchor-free mechanism significantly reduces the number of design parameters which need heuristic tuning and many tricks involved for good performance, making the detector, especially its training and decoding phase, considerably simpler.
- TO switch YOLO into an anchor-free manner is quite simple, authors reduce the predictions for each location from 3 to 1 and make them directly predict four values, i.e., two offsets in terms of the left-top corner of the grid, and the height and width of the predicted box. They assign the center location of each object as the positive sample and pre-define a scale range to designate the FPN level for each object. Such modification reduces the parameters and GFLOPs of the detector and makes it faster, but obtains better performance.

(3) SimOTA

Authors simplify OTA to dynamic top-k strategy, named SimOTA, which not only reduces the training time but also avoids additional solver hyperparameters in SinkhornKnopp algorithm. 
> The paper gives a brief introduction to simota, and OTA's paper(https://arxiv.org/abs/2103.14259) explains its principle in more detail.

4.どうやって有効だと検証した? /How is this work validated?

- Table 2 shows that with YOLOv3 as the baseline, all the methods the authors tried and the AP improvements they achieved. Some of them have similar ideas in YOLOv5.

![Table 2](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/2e42565d-8e6a-4a74-a566-4318fae7a443)

- Besides DarkNet53, the authors also test YOLOX on other backbones with different sizes, where YOLOX achieves consistent improvements against all the corresponding counterparts.

![Table 3](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/29b94179-a930-4811-88b2-5369a5b70a48)
![Table 4](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/68954a52-9a1d-4e24-81de-97b102b8dc3d)

5.議論はある? /Any discussion for this work?

Figure 1では、なぜリュオンモデルと比較しているのかがありますが、縦軸はパラメーターであり、Latencyではありません。

6.読んでいてわからなかったところは? /What don't you understand for this paper?

7.公開コードやデータセットは? /Public codes or datasets?

https://github.com/Megvii-BaseDetection/YOLOX

8.次に読むべき論文は? /What should we read next?
